File: clients/javascript/model/loot.js
const Item = require.main.require('./model/item');
const Vec2 = require.main.require('./model/vec2');
/**
 * Loot lying on the ground
 */
class Loot {
    /**
     * Unique id
     */
    id;
    /**
     * Position
     */
    position;
    /**
     * Item
     */
    item;

    constructor(id, position, item) {
        this.id = id;
        this.position = position;
        this.item = item;
    }

    /**
     * Read Loot from input stream
     */
    static async readFrom(stream) {
        let id;
        id = await stream.readInt();
        let position;
        position = await Vec2.readFrom(stream);
        let item;
        item = await Item.readFrom(stream);
        return new Loot(id, position, item);
    }

    /**
     * Write Loot to output stream
     */
    async writeTo(stream) {
        let id = this.id;
        await stream.writeInt(id);
        let position = this.position;
        await position.writeTo(stream);
        let item = this.item;
        await item.writeTo(stream);
    }
}
module.exports = Loot

File: clients/python/model/loot.py
from model.item import Item
from model.vec2 import Vec2
from stream_wrapper import StreamWrapper

class Loot:
    """Loot lying on the ground"""

    __slots__ = ("id","position","item",)

    id: int
    position: Vec2
    item: Item

    def __init__(self, id: int, position: Vec2, item: Item):
        self.id = id
        """Unique id"""
        self.position = position
        """Position"""
        self.item = item
        """Item"""

    @staticmethod
    def read_from(stream: StreamWrapper) -> "Loot":
        """Read Loot from input stream
        """
        id = stream.read_int()
        position = Vec2.read_from(stream)
        item = Item.read_from(stream)
        return Loot(id, position, item)
    
    def write_to(self, stream: StreamWrapper):
        """Write Loot to output stream
        """
        stream.write_int(self.id)
        self.position.write_to(stream)
        self.item.write_to(stream)
    
    def __repr__(self):
        return "Loot(" + \
            repr(self.id) + \
            ", " + \
            repr(self.position) + \
            ", " + \
            repr(self.item) + \
            ")"